### PR TITLE
[(HTTP)URLResponse] NSCoding implementation

### PR DESCRIPTION
**Pull request by naithar**
Implemented `NSCoding` protocol for `URLResponse` ant its subclass `HTTPURLResponse`.
Added tests.

### CODE REVIEW

**Comment by naithar** [comment]
@parkera can you review this?

File: TestFoundation/TestNSURLResponse.swift [hunk 1]
```diff
@@ -91,6 +92,19 @@ class TestNSURLResponse : XCTestCase {
         let res = URLResponse(url: url, mimeType: "txt", expectedContentLength: 0, textEncodingName: nil)
         XCTAssertTrue(res.isEqual(res.copy() as! NSObject))
     }
+    
+    func test_NSCoding() {
+        let url = URL(string: "https://apple.com")!
+        let responseA = URLResponse(url: url, mimeType: "txt", expectedContentLength: 0, textEncodingName: nil)
+        let responseB = NSKeyedUnarchiver.unarchiveObject(with: NSKeyedArchiver.archivedData(withRootObject: responseA)) as! URLResponse
+        
+        //On macOS unarchived Archived then unarchived `URLResponse` is not equal.
```
Comment (parkera) [inline TestFoundation/TestNSURLResponse.swift]: This is interesting and probably worth a bug for us.

**Comment by parkera** [comment]
Thanks @naithar!

**Comment by parkera** [comment]
@swift-ci please test and merge

**Pull request merged**